upgrade/1.2/scripts/common/upgrade-state.sh

```diff
@@ -98,6 +98,12 @@ function err_report() {
     if [[ ${shouldIgnore} -eq 1 ]]; then
         return 0
     fi
+
+    ignoreCmd="grep -v -e Running -e Completed > /dev/null"
```

Comment:
I'm not following how we think that line is tripping an error because there is a `set +e` and `set -e` around that set of lines.    Could you explain more?

Reply:
our set +e will make sure it doesn't exist but it is caught by our `trap 'err_report' ERR`. and this error is coming from our trap function

Reply:
Okay, I see that know.   It seems like there is a better way to handle this.  Looking.

Reply:
Okay, how about 
```
echo "$output" | grep -v -e Running -e Completed > /dev/null || rc=$?
```
instead of 
```
echo "$output" | grep -v -e Running -e Completed > /dev/null
rc=$?
```

Reply:
That will make the command always have a non-zero return code but still set the rc for the grep part of the command.

Reply:
that is better.